src/librustc_mir/interpret/operand.rs

```diff
@@ -472,39 +472,37 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     // avoid allocations.
     pub(super) fn eval_place_to_op(
         &self,
-        mir_place: &mir::Place<'tcx>,
+        place: &mir::Place<'tcx>,
         layout: Option<TyLayout<'tcx>>,
     ) -> InterpResult<'tcx, OpTy<'tcx, M::PointerTag>> {
         use rustc::mir::PlaceBase;
 
-        mir_place.iterate(|place_base, place_projection| {
-            let mut op = match place_base {
-                PlaceBase::Local(mir::RETURN_PLACE) =>
-                    throw_unsup!(ReadFromReturnPointer),
-                PlaceBase::Local(local) => {
-                    // Do not use the layout passed in as argument if the base we are looking at
-                    // here is not the entire place.
-                    // FIXME use place_projection.is_empty() when is available
-                    let layout = if mir_place.projection.is_none() {
-                        layout
-                    } else {
-                        None
-                    };
-
-                    self.access_local(self.frame(), *local, layout)?
-                }
-                PlaceBase::Static(place_static) => {
-                    self.eval_static_to_mplace(place_static)?.into()
-                }
-            };
+        let mut op = match &place.base {
```

Comment:
When Place 2.0 is done, will we be able to get rid of the mutability here again?
I think there was no mutability when this started, and it would be nice to be able to get back to clean code with immutable variables.

Or maybe this is just a matter of rewriting the final `for elem in place.projection.iter()` into a fold?

Reply:
It's a matter of writing it using fold, gonna send a PR about it as soon as I can resync and compile my rustc copy.

Reply:
Hmmm in order to avoid the `mut` there I'd need to desugar the `?` operator, because it won't work inside `fold`. Do you think it worth? any better idea?.

Reply:
Thanks for taking a look!

Does [`try_fold`](https://doc.rust-lang.org/nightly/std/iter/trait.Iterator.html#method.try_fold) help?

Reply:
Yes, that's what I needed https://github.com/rust-lang/rust/pull/64473